packages/audiofileplayer/android/src/main/java/com/google/flutter/plugins/audiofileplayer/AudiofileplayerPlugin.java

```diff
@@ -112,37 +121,103 @@
   // Used when defining an Intent from a custom media button.
   public static final String CUSTOM_MEDIA_BUTTON_EXTRA_KEY = "customMediaButton";
 
-  private final Registrar registrar;
-  private final Map<String, ManagedMediaPlayer> mediaPlayers;
-  private final MethodChannel methodChannel;
+  private Registrar registrar;
+  private Activity activity;
+  private Map<String, ManagedMediaPlayer> mediaPlayers;
+  private MethodChannel methodChannel;
+  private Context context;
+  private ActivityPluginBinding activityPluginBinding;
 
```

Comment:
You don't need to keep the binding here. Looks like you only need activity. So set Activity object for v2. Get activity from registrar for v1. See my other comment.

Reply:
Done